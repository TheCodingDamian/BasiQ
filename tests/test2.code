func main() -> void
{
    cnt: num = 0
    while cnt < 15
    {
        println("------------------------------------------")
        println(cnt + "! = " + factorial(cnt)))
        println("fib_" + cnt + " = " + fibonacci(cnt)))
        println("PI with accuracy " + cnt*10 + " = " + compute_pi(cnt*10))
        println("------------------------------------------")
        println("")
        cnt = cnt + 1
    }

    fizzbuzz(100)
}

func fizzbuzz(max: num) -> void
{
    cnt: num = 0
    while cnt < max:
    {
        cnt = cnt + 1
        if cnt % 15 == 0
            println("FIZZBUZZ")
        else if cnt % 3 == 0
            println("FIZZ")
        else if cnt % 5 == 0
            println("BUZZ")
        else
            println(to_text(cnt))
    }
}

func factorial(n: num) -> num
{
    if n < 0
    {
        return 0 - 1
    }
    if n == 0
    {
        return 1
    }
    return n * factorial(n - 1)
}

func fibonacci(n: num) -> num
{
    if n == 0
    {
        return 1
    }
    if n == 1
    {
        return 1
    }

    f1: num = 1
    f2: num = 1

    cnt: num = 1
    while cnt < n
    {
        cnt = cnt + 1
        temp: num = f1 + f2
        f1 = f2
        f2 = temp
    }
    return f2
}

func compute_pi(n: num) -> num
{
    sum: num = 0
    i: num = 0

    while i <= n
    {
        sum = sum + ((-1)**(i % 2)) * (1 / (1 + 2 * i))
        i = i + 1
    }

    return sum * 4
}

main()