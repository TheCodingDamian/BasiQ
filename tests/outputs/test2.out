------------------------------------------
0! = 1
fib_0 = 1
PI with accuracy 0 = 4
------------------------------------------

------------------------------------------
1! = 1
fib_1 = 1
PI with accuracy 10 = 3.232315809405594
------------------------------------------

------------------------------------------
2! = 2
fib_2 = 2
PI with accuracy 20 = 3.189184782277596
------------------------------------------

------------------------------------------
3! = 6
fib_3 = 3
PI with accuracy 30 = 3.1738423371907505
------------------------------------------

------------------------------------------
4! = 24
fib_4 = 5
PI with accuracy 40 = 3.1659792728432157
------------------------------------------

------------------------------------------
5! = 120
fib_5 = 8
PI with accuracy 50 = 3.1611986129870506
------------------------------------------

------------------------------------------
6! = 720
fib_6 = 13
PI with accuracy 60 = 3.157984995168666
------------------------------------------

------------------------------------------
7! = 5040
fib_7 = 21
PI with accuracy 70 = 3.155676462307475
------------------------------------------

------------------------------------------
8! = 40320
fib_8 = 34
PI with accuracy 80 = 3.1539378622726155
------------------------------------------

------------------------------------------
9! = 362880
fib_9 = 55
PI with accuracy 90 = 3.1525813328751204
------------------------------------------

------------------------------------------
10! = 3628800
fib_10 = 89
PI with accuracy 100 = 3.1514934010709914
------------------------------------------

------------------------------------------
11! = 39916800
fib_11 = 144
PI with accuracy 110 = 3.150601479819498
------------------------------------------

------------------------------------------
12! = 479001600
fib_12 = 233
PI with accuracy 120 = 3.149856975293274
------------------------------------------

------------------------------------------
13! = 6227020800
fib_13 = 377
PI with accuracy 130 = 3.1492261301786892
------------------------------------------

------------------------------------------
14! = 87178291200
fib_14 = 610
PI with accuracy 140 = 3.1486847629938386
------------------------------------------

1
2
FIZZ
4
BUZZ
FIZZ
7
8
FIZZ
BUZZ
11
FIZZ
13
14
FIZZBUZZ
16
17
FIZZ
19
BUZZ
FIZZ
22
23
FIZZ
BUZZ
26
FIZZ
28
29
FIZZBUZZ
31
32
FIZZ
34
BUZZ
FIZZ
37
38
FIZZ
BUZZ
41
FIZZ
43
44
FIZZBUZZ
46
47
FIZZ
49
BUZZ
FIZZ
52
53
FIZZ
BUZZ
56
FIZZ
58
59
FIZZBUZZ
61
62
FIZZ
64
BUZZ
FIZZ
67
68
FIZZ
BUZZ
71
FIZZ
73
74
FIZZBUZZ
76
77
FIZZ
79
BUZZ
FIZZ
82
83
FIZZ
BUZZ
86
FIZZ
88
89
FIZZBUZZ
91
92
FIZZ
94
BUZZ
FIZZ
97
98
FIZZ
BUZZ